systems/lanl-venado/system.py

```diff
@@ -0,0 +1,214 @@
+# Copyright 2023 Lawrence Livermore National Security, LLC and other
+# Benchpark Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import pathlib
+
+from benchpark.directives import variant
+from benchpark.system import System
+
+id_to_resources = {
+    "grace-hopper": {
+        "sys_cores_per_node": 144,
+        "sys_gpus_per_node": 4,
+    },
+    "grace-grace": {
+        "sys_cores_per_node": 144,
+    },
+}
+
+
+class LanlVenado(System):
+    variant(
+        "cluster",
+        default="grace-hopper",
+        values=("grace-hopper", "grace-grace"),
+        description="Which cluster to run on",
+    )
+
+    variant(
+        "cuda",
+        default="12-5",
+        values=("11.8", "12.5"),
+        description="CUDA version",
+    )
+
+    variant(
+        "compiler",
+        default="cce",
+        values=("gcc", "cce"),
+        description="Which compiler to use",
+    )
+
+    variant(
+        "gtl",
+        default=False,
+        values=(True, False),
+        description="Use GTL-enabled MPI",
+    )
+
+    variant(
+        "lapack",
+        default="cusolver",
+        values=("cusolver", "cray-libsci"),
+        description="Which lapack to use",
+    )
+
+    variant(
+        "blas",
+        default="cublas",
+        values=("cublas", "cray-libsci"),
+        description="Which blas to use",
+    )
+
+    def initialize(self):
+        super().initialize()
+
+        self.scheduler = "slurm"
+        attrs = id_to_resources.get(self.spec.variants["cluster"][0])
+        for k, v in attrs.items():
+            setattr(self, k, v)
+
+    def generate_description(self, output_dir):
+        super().generate_description(output_dir)
+
+        sw_description = pathlib.Path(output_dir) / "software.yaml"
+
+        with open(sw_description, "w") as f:
+            f.write(self.sw_description())
+
+    def system_specific_variables(self):
+        return {
+            "cuda_arch": "90",
+            "default_cuda_version": self.spec.variants["cuda"][0],
+            "extra_batch_opts": '"-A llnl_ai_g -pgpu"',
```

Comment:
@rfhaque what does this extra_batch_opts do?

Reply:
@pearce8 The extra_batch_opts requests the required partition and account. We may need to parameterize this